__tests__/internals/network/networkHelpers.test.js

```diff
@@ -8,16 +8,31 @@ describe('getNetworkHelpers', () => {
     expect(getNetworkHelpers()).toMatchSnapshot();
   });
 
-  test('after setting them', () => {
+  test('after setting them', async () => {
     const tokenName = 'customToken';
 
     setNetworkHelpers({
       getToken: () => tokenName,
     });
 
     expect(getNetworkHelpers().getToken()).toBe(tokenName);
-    expect(getNetworkHelpers().requestGET().headers.Authorization).toBe(
+    expect((await getNetworkHelpers().requestGET()).headers.Authorization).toBe(
       `Bearer ${tokenName}`,
     );
   });
+
+
+  test('after setting them - promise token', async () => {
+    const tokenName = 'customToken';
+
+    setNetworkHelpers({
+      getToken: () => Promise.resolve(tokenName),
```

Comment:
I think this line could be simplified to `getToken: async () => tokenName,` to highlight the asynchronicity of `getToken`